### PR TITLE
Greyscale cables

### LINKED ISSUE #29821: Cable coils don't produce the correct cable cuff subtype for it's color
Repo:

1. Get a stack of cyan cabling (either /obj/item/stack/cable_coil/random or /obj/item/stack/cable_coil/cyan)
2. Make cable cuffs with said stack of cable
3. end up with red ones (/obj/item/weapon/restraints/handcuffs/cable) instead of the properly colored subtype, /obj/item/weapon/restraints/handcuffs/cable/cyan

This is true for all colors of cable.

Was planning on adding proper inhand sprites for each color variant, but there isn't much point of they aren't actually working anyway.

### PR DESCRIPTION
**Pull request by ninjanomnom**
image

:cl: ninjanomnom & Wjohn
fix: Cable cuffs now inherit the color of the cables used to make them.
fix: Split cable stacks keep their color.
tweak: The blue cable color is now a better blue.
/:cl:

fixes #29821 

Thanks to @WJohn for the sprites

I probably missed some places where color is set, the cable code is labyrinthine and this took longer than I expected as a result.

### CODE REVIEW

**Comment by Anonmare** [comment]
You're playing with powers you don't understand mortal

**Comment by ChangelingRain** [comment]
That is nooot the current blue at all. I think you need some tuning.
And a picture that isn't super horrible, like jeez.

**Comment by ninjanomnom** [comment]
@Anonmare Coders aren't mortal, our contributions just slow down and we exist forever in the code.

**Comment by ninjanomnom** [comment]
Also I hated the old blue, it was a gross color.

**Comment by Bawhoppen** [comment]
what icon scale do you have on jeez

**Comment by praisenarsie** [comment]
Did you fix where if you take ONE wire from a cable coil, it turns into a completely different set color and it has the cable coil sprite rather than the cable piece sprite.

**Comment by MoreRobustThanYou** [comment]
dude take your pictures with 64x64 scale, else they look like shit

**Comment by MoreRobustThanYou** [comment]
@ninjanomnom don't forget to update the RCL color changing function

**Comment by ninjanomnom** [comment]
@MoreRobustThanYou I don't need to, cables support both string and rgb values for color

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 1]
```diff
@@ -490,7 +490,7 @@ Congratulations! You are now trained for invasive xenobiology research!"}
 /obj/item/restraints/handcuffs/energy
 	name = "hard-light energy field"
 	desc = "A hard-light field restraining the hands."
-	icon_state = "cuff_white" // Needs sprite
+	icon_state = "cuff" // Needs sprite
```
Comment (nicbn) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm]: Take that comment off

File: code/game/objects/items/handcuffs.dm [hunk 2]
```diff
@@ -98,6 +99,21 @@
 	cuffsound = 'sound/weapons/cablecuff.ogg'
 	var/datum/robot_energy_storage/wirestorage = null
 
+/obj/item/restraints/handcuffs/cable/Initialize(mapload, color_override)
+	. = ..()
+
+	var/list/cable_colors = GLOB.cable_colors
+	if(color_override)
```
Comment (nicbn) [inline code/game/objects/items/handcuffs.dm]:
```
item_color = color_override || item_color || pick(cable_colors)
if(cable_colors[cable_color])
	cable_color = cable_colors[cable_color]
```

Choose to use either color_override, override_color or param_color and don't use other ones.

File: code/modules/power/cable.dm [hunk 3]
```diff
@@ -87,6 +89,15 @@ By design, d1 is the smallest direction and d2 is the highest
 	else
 		stored = new/obj/item/stack/cable_coil(null,1,cable_color)
 
+	var/list/cable_colors = GLOB.cable_colors
+	if(override_color)
```
Comment (nicbn) [inline code/modules/power/cable.dm]:
```
item_color = override_color || item_color || pick(cable_colors)
if(cable_colors[cable_color])
	cable_color = cable_colors[cable_color]
```

File: code/modules/power/cable.dm [hunk 4]
```diff
@@ -497,8 +489,15 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list (new/datum/stack_recipe("cable restrai
 	. = ..()
 	if(new_amount) // MAXCOIL by default
 		amount = new_amount
+
+	var/list/cable_colors = GLOB.cable_colors
 	if(param_color)
```
Comment (nicbn) [inline code/modules/power/cable.dm]:
```
item_color = param_color || item_color || pick(cable_colors)
if(cable_colors[item_color])
	item_color = cable_colors[item_color]
```

**Comment by nicbn** [comment]
Also choose string or RGB as color parameter and don't allow the other one to be used.

File: code/modules/power/cable.dm [hunk 5]
```diff
@@ -1,3 +1,14 @@
+GLOBAL_LIST_INIT(cable_colors, list(
+	"yellow" = rgb(255, 255, 0),
+	"green" = rgb(0, 170, 0),
+	"blue" = rgb(25, 25, 200),
+	"pink" = rgb(255, 60, 200),
+	"orange" = rgb(255, 128, 0),
+	"cyan" = rgb(0, 255, 255),
+	"white" = rgb(255, 255, 255),
+	"red" = rgb(255, 0, 0)
```
Comment (ChangelingRain) [inline code/modules/power/cable.dm]: make these hex color codes, please

**Comment by ShizCalev** [comment]
Would this close #29821?

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/game/gamemodes/miniantags/abduction/abduction_gear.dm]: Ehh, it still applies here

File: code/modules/power/cable.dm (hunk 3 above)
Comment (ShizCalev) [reply code/modules/power/cable.dm]: malkovich malkovich

**Comment by ninjanomnom** [comment]
No, someone else was working on that @ShizCalev

**Comment by kingofkosmos** [comment]
>No, someone else was working on that

That was me, but tbh it's too difficult. I'm hoping you could fix it with this one.

**Comment by ninjanomnom** [comment]
Alright, I'll take a look at it then

**Comment by AnonymousNow** [comment]
Looks very nice, especially the significantly better blue.

**Comment by WJohn** [comment]
Note that I tweaked the lighting for basically every cable so the light source comes consistently from the top left (shadows are always under and on the right side of cables). This should fix longstanding issues of lighting just spontaneously swapping from above to below in a lot of places. If you spot any more of these, let me know, I may have goofed here or there.

**Comment by praisenarsie** [comment]
bugs:

vendors always dispense red cable coils like the YouTool
SecTech vendor dispenses red cable restraint instead of regular white zipties
If you take a cable from a cable coil stack, it DOES NOT give the CABLE PIECE SPRITE

**Comment by ninjanomnom** [comment]
Those have been fixed now

**Comment by PKPenguin321** [comment]
>an icon change that is an improvement

rare pr

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm (hunk 1 above)
Comment (nicbn) [reply code/game/gamemodes/miniantags/abduction/abduction_gear.dm]: But it is obvious

**Pull request merged**